pkg/databricks/client.go

```diff
@@ -42,18 +44,25 @@ type Client struct {
 	isWSAPIAvailable  bool
 }
 
-func NewClient(ctx context.Context, httpClient *http.Client, hostname, accountHostname, accountID string, auth Auth) (*Client, error) {
-	if hostname == "" {
-		hostname = defaultHost
+func GetHostname(cfg *viper.Viper) string {
+	if cfg.GetString(config.HostnameField.FieldName) == "" {
+		return defaultHost
 	}
+	return cfg.GetString(config.HostnameField.FieldName)
+}
+
+func GetAccountHostname(cfg *viper.Viper) string {
+	if cfg.GetString(config.AccountHostnameField.FieldName) == "" {
+		return "accounts." + GetHostname(cfg)
```

Comment:
I don't think this will work for Azure or GCP, as the hostname provided will be something like `8757561887652360.0.gcp.databricks.com` or `adb-5555555555555555.19.azuredatabricks.net`, but the account hostnames for those need to be `accounts.gcp.databricks.com` and `accounts.azuredatabricks.net` respectively. Prepending "accounts" to the hostnames will result in `accounts.8757561887652360.0.gcp.databricks.com` and `accounts.adb-5555555555555555.19.azuredatabricks.net`, which isn't correct.

Reply:
@ggreer Isn't that solved by account-hostname ?

Reply:
Yes but we should change the behavior to detect GCP and Azure hostnames and return correct account hostnames for them.

Reply:
@ggreer I made this change https://github.com/ConductorOne/baton-databricks/pull/15/commits/7de479121ec92bcc651e5e763e71995c264dee34